packages/client/src/runtime/core/types/Extensions.ts

```diff
@@ -21,30 +25,178 @@ export type Args = InternalArgs
 export type DefaultArgs = InternalArgs<{}, {}, {}, {}>
 
 export type GetResult<Base extends Record<any, any>, R extends Args['result'][string], _R extends Args['result'][string] = Record<string, any> extends R ? {} : R> =
-  { [K in keyof _R | keyof Base]: K extends keyof _R ? ReturnType<ReturnType<_R[K]>['compute']> : Base[K] } & unknown
+  { [K in keyof _R | keyof Base]: K extends keyof _R ? Return<Path<Return<_R[K]>, ['compute']>> : Base[K] }
 
 export type GetSelect<Base extends Record<any, any>, R extends Args['result'][string], _R extends Args['result'][string] = Record<string, any> extends R ? {} : R> =
   { [K in keyof _R | keyof Base]?: K extends keyof _R ? boolean : Base[K] }
 
-export type GetModel<Base extends Record<any, any>, M extends Args['model'][string], _M extends Args['model'][string] = Record<string, any> extends M ? {} : M> =
-  { [K in keyof _M | keyof Base]: K extends keyof _M ? ReturnType<_M[K]> : Base[K] }
+/** Query */
+
+export type DynamicQueryExtensionArgs<Q_, TypeMap extends Record<string, any>> = {
+  [K in keyof Q_]:
+    K extends '$allOperations'
+    ? (args: { model?: string, operation: string, args: any, query: (args: any) => PrismaPromise<any> }) => Promise<any>
```

Comment:
So, for top-level `$allOperations` you are not getting inference on model/args, am I getting this right? Should we?

Reply:
I'll update it.

Reply:
So in some cases the union can get pretty crazy and I am not sure if this would be useful in terms of performance. I am not sure why the compiler is able to handle `$.allModels.$allOperations` though. I think it has to do with the share of args being a record always and now we'd be mixing arrays and classes potentially, so the intersection of that on access of a property can be pretty wild - well, that's my guess. So I am leaving it as is for now.